src/casscf/casscf_gradient.cc

```diff
@@ -0,0 +1,477 @@
+/**
+ * CASSCF gradient code by Shuhe Wang
+ *
+ * The computation procedure is listed as belows:
+ * (1), Set MOs spaces;
+ * (2), Set Tensors (F, H, V etc.); 
+ * (3), Compute and write the Lagrangian;
+ * (4), Write 1RDMs and 2RDMs coefficients;
+ * (5), Back-transform the TPDM. 
+ */
+
+#include "casscf/casscf.h"
+#include "gradient_tpdm/backtransform_tpdm.h"
+#include "ambit/tensor.h"
+#include "ambit/blocked_tensor.h"
+
+#include "integrals/integrals.h"
+#include "base_classes/rdms.h"
+
+#include "psi4/libfock/jk.h"
+#include "psi4/libmints/matrix.h"
+#include "psi4/libmints/wavefunction.h"
+#include "psi4/libqt/qt.h"
+#include "psi4/psifiles.h"
+
+#include "helpers/printing.h"
+#include "helpers/helpers.h"
+#include "helpers/timer.h"
+#include "sci/aci.h"
+#include "fci/fci_solver.h"
+#include "base_classes/active_space_solver.h"
+
+#include "sci/fci_mo.h"
+#include "orbital-helpers/mp2_nos.h"
+#include "orbital-helpers/orbitaloptimizer.h"
+#include "orbital-helpers/semi_canonicalize.h"
+
+#ifdef HAVE_CHEMPS2
```

Comment:
Please clean up the headers and clang-format the file.

Reply:
The header has been cleaned.